testutils/consul.go

```diff
@@ -144,6 +210,67 @@ func DeleteKV(tb testing.TB, srv *testutil.TestServer, key string) {
 	defer resp.Body.Close()
 }
 
+type Check struct {
+	CheckID     string
+	Name        string
+	Status      string
+	Notes       string
+	Output      string
+	ServiceID   string
+	ServiceName string
+	ServiceTags []string
+	Type        string
+}
+
+func ListChecks(tb testing.TB, srv *testutil.TestServer) map[string]Check {
+	u := fmt.Sprintf("http://%s/v1/agent/checks", srv.HTTPAddr)
+	resp := RequestHTTP(tb, http.MethodGet, u, "")
+
+	b, err := ioutil.ReadAll(resp.Body)
+	require.NoError(tb, err)
+	defer resp.Body.Close()
+
+	require.Equal(tb, resp.StatusCode, 200, string(b))
+
+	var checks map[string]Check
+	err = json.Unmarshal(b, &checks)
+	require.NoError(tb, err)
+	return checks
+}
+
+func WaitForCheckStatus(tb testing.TB, srv *testutil.TestServer, checkID, status string, wait time.Duration) (Check, error) {
```

Comment:
Something I've noted is that we have a lot of WaitFor methods that all have the same structure. I'd like to refactor this in the future so that they all use a common method for polling, kind of in the same way we have a generic retry structure in the CTS code. I'll be making an issue for this.

Reply:
Opened https://github.com/hashicorp/consul-terraform-sync/issues/859 to address this.